examples/amm/src/lib.rs

```diff
@@ -0,0 +1,124 @@
+use std::convert::Infallible;
+
+use async_graphql::{scalar, Request, Response};
+use fungible::AccountOwner;
+use linera_sdk::base::{Amount, ArithmeticError, ContractAbi, ServiceAbi};
+use linera_views::views::ViewError;
+use matching_engine::Parameters;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+pub struct AmmAbi;
+
+impl ContractAbi for AmmAbi {
+    type InitializationArgument = ();
+    type Parameters = Parameters;
+    type Operation = Operation;
+    type ApplicationCall = ApplicationCall;
+    type Message = Message;
+    type SessionCall = ();
+    type Response = ();
+    type SessionState = ();
+}
+
+impl ServiceAbi for AmmAbi {
+    type Query = Request;
+    type QueryResponse = Response;
+    type Parameters = Parameters;
+}
+
+/// Operations that can be sent to the application.
+#[derive(Debug, Serialize, Deserialize)]
+pub enum Operation {
+    Swap {
```

Comment:
This would be the right place to document what the operations are doing and what the arguments mean. In particular, `owner` means something different in each operation.